fmt/testdata/LiteralExpression/original.sol

```diff
@@ -0,0 +1,47 @@
+contract LiteralExpressions {
+    function test() external {
+        // bool literals
+        true;
+        false;
+        /* comment1 */ true /* comment2 */;
+        // comment3
+        false; // comment4
+
+        // number literals
+        1;
+        123_000;
+        1_2e345_678;
+        -1;
+        2e-10;
+        // comment5
+        /* comment6 */ -1 /* comment7 */;
+
+        // hex number literals
+        0x00;
+        0x123_456;
+        0x2eff_abde;
+
+        // rational number literals
+        .1;
+        1.3;
+        2.5e1;
+
+        // string literals
+        "";
+        "foobar";
+        "foo" // comment8
+            " bar";
+        // comment9
+        "\
+some words" /* comment10 */;
```

Comment:
this is the line where the test currently fails. looking for a way to bypass any indentation rn. @jpopesculian @mattsse do you have any ideas? it seems like `write_raw` still indents by the line by base

Reply:
cc @seanyoung too in case helpful

Reply:
the indentation. unfortunately it is not displayed correctly in the tests, but I'm looking for a way to bypass all of the buffers to be able to write without any indentation.
given code
```solidity
function test() external {
        "\
some words" /* comment10 */;
}
```
it is formatted to 
```solidity
function test() external {
        "\
    some words" /* comment10 */;
}

Reply:
@rkrasiuk i think to solve this we'll have to make the IndentWriter context aware and write string literals directly to the underlying writer. i'll open a PR hopefully today, definitely by tomorrow morning (i have to fly today) to this branch